### ISSUE TITLE
Can't build RocksDB 3.8 for Java

### ISSUE THREAD

**Issue by criccomini** [facebook/rocksdb#418]
When running crossbuilds for RocksDB 3.8 JARs against Vagrant's CentOS VM, I get:

```
==> linux32: util/options_builder.cc: In function ‘void rocksdb::{anonymous}::OptimizeForLevel(int, int, uint64_t, rocksdb::Options*)’:
==> linux32: util/options_builder.cc:164:23: error: comparison between signed and unsigned integer expressions [-Werror=sign-compare]
==> linux32: g++ -g -Wall -Werror -Wsign-compare -Wshadow -I. -I./include -std=c++11  -DROCKSDB_PLATFORM_POSIX  -DOS_LINUX -fno-builtin-memcmp -DGFLAGS=google -DZLIB -DBZIP2   -DHAVE_JEMALLOC -fPIC -DNDEBUG -O2 -O2 -fno-omit-frame-pointer -momit-leaf-frame-pointer -Woverloaded-virtual -Wnon-virtual-dtor -c util/options_helper.cc -o util/options_helper.o 
==> linux32: cc1plus: all warnings being treated as errors
==> linux32: make[1]: 
==> linux32: *** [util/options_builder.o] Error 1
==> linux32: make[1]: 
==> linux32: *** Waiting for unfinished jobs....
==> linux32: make[1]: Leaving directory `/rocksdb'
==> linux32: make: 
==> linux32: *** [rocksdbjavastatic] Error 2
```

I am unable to release RocksDB 3.8 because of this.

This can be reproduced by running:

```
make jclean clean rocksdbjavastaticpublish
```

**Comment by fyrz** [comment]
@criccomini did you find the time to check 3.9.0 ?

**Comment by criccomini** [comment]
@fyrz Just released 3.9.0. Any idea when the next version is going to be released? Personally interested in the TTL Java patch that was applied mid-Jan.

**Comment by igorcanadi** [comment]
Should be soon :) Maybe week or two

**Comment by fyrz** [comment]
@criccomini is there any way to republish 3.9.0 ? I suppose that you build the rocksdbjava library upon tag 3.9 ? Sadly the thing is that this tag is missing a commit which is committed to the 3.9.fb branch. What leads to the situation that the RocksDB jar will fail with Sigsegv under certain circumstances. Especially while using read-only mode whats the only way to use a RocksDB database by multiple processes.

@igorcanadi @yhchiang  how can we handle this in future ? It shouldn`t happen that 3.9.fb is one commit ahead to the tag without having a patch tag right ?

@criccomini can you please verify before publishing with `jtest` target in future ?

**Comment by siying** [comment]
@fyrz we can change tag to point to latest build of the branch. Will that solve the problem?

**Comment by igorcanadi** [comment]
Oh wow, this is a serious issue. I'll publish RocksDB 3.9.1 with the missing commit. This was a mistake on our part.

**Comment by igorcanadi** [comment]
@siying Releases are supposed to be immutable, so we'll need to create release 3.9.1.

**Comment by igorcanadi** [comment]
Removed 3.9 and released 3.9.1 with the fix.

**Comment by criccomini** [comment]
@fyrz please add jtest as a dependency to the jni publication target. This way it'll get run automatically.

While you're at it, add this as the JAVA_INCLUDE for Mac, as well:

            JAVA_INCLUDE = -I/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.9.sdk/System/Library/Frameworks/JavaVM.framework/Versions/A/Headers

**Comment by criccomini** [comment]
Published 3.9.1. `make jtest` passed.
